### PR TITLE
First project info

### PR DESCRIPTION
**Pull request by simon5656**

### CODE REVIEW

**Comment by DBees** [comment]
@simon5656 as agreed the monthly reports are to live in https://github.com/openhwgroup/program/TGs/verification-task-group/Monthly-Reports

Please don't put them here

**Comment by DBees** [comment]
As mentioned above the project reports don't belong in this repo, they should go in the programs repo

**Comment by simon5656** [comment]
as I have said - when you have sorted out the structure of the programs repo I will move them - please don't delay this repo until you sort the other one - please merge this now.

**Comment by DBees** [comment]
Hi Simon

It is created as we discussed via merged pull request. The structure is there… down to  <https://github.com/openhwgroup/programs> programs/ <https://github.com/openhwgroup/programs/tree/master/TGs> TGs/ <https://github.com/openhwgroup/programs/tree/master/TGs/verification-task-group> verification-task-group/Project-Reports/ and  <https://github.com/openhwgroup/programs/tree/master/TGs/verification-task-group/Meetings> https://github.com/openhwgroup/programs/tree/master/TGs/verification-task-group/Meetings

 

However as there is also legacy stuff in there, please work with Mike to sort out what legacy stuff should be kept and what should be folded into which folder, keeping mind the agreed structure. Or if you both want to meet with me and sort through it together, I’m happy to do the cleanup. I would strongly prefer that we put the new files into the new agreed structure to start in the right direction. 

Duncan

 



 

You’ll note within the verification-task-group there are several extraneous folders from before. Please decide with Mike what to keep and what to chuck away or how to sort those.

 

I would really rather not merge into the other repo as I’m pretty sure once we start one way it will be harder to

**Comment by simon5656** [comment]
updated with second data sets

**Comment by simon5656** [comment]
updated to add zfinx

**Review by MikeOpenHWGroup — CHANGES_REQUESTED** [review]
Please remove [2022-Aug10-ARVM-Advanced-riscv-Verification-Methodologies-OpenHW-Project-Concept-Proposal.md](ARVM-TopProject/2022-Aug10-ARVM-Advanced-riscv-Verification-Methodologies-OpenHW-Project-Concept-Proposal.md) as this is already in `openhwgroup/programs`.

File: ARVM-TopProject/2022-Aug10-ARVM-Advanced-riscv-Verification-Methodologies-OpenHW-Project-Concept-Proposal.md
Comment (MikeOpenHWGroup) [inline ARVM-TopProject/2022-Aug10-ARVM-Advanced-riscv-Verification-Methodologies-OpenHW-Project-Concept-Proposal.md]: Please delete this file (already in `openhwgroup/programs`).

File: ARVM-TopProject/README.md
Comment (MikeOpenHWGroup) [inline ARVM-TopProject/README.md]: Point to `openhwgroup/programs`.

**Comment by simon5656** [comment]
resolved - pls merge

File: ARVM-TopProject/2022-Aug10-ARVM-Advanced-riscv-Verification-Methodologies-OpenHW-Project-Concept-Proposal.md
Comment (MikeOpenHWGroup) [reply ARVM-TopProject/2022-Aug10-ARVM-Advanced-riscv-Verification-Methodologies-OpenHW-Project-Concept-Proposal.md]: Thanks

File: ARVM-TopProject/README.md
Comment (MikeOpenHWGroup) [reply ARVM-TopProject/README.md]: Thanks

**Review by MikeOpenHWGroup — APPROVED** [review]
LGTM, thanks @simon5656.   @DBees, let me know if you have any outstanding items.

**Review by MikeOpenHWGroup — APPROVED** [review]

**Review by MikeOpenHWGroup — CHANGES_REQUESTED** [review]
The coverage log files (e.g. `ARVM-FunctionalCoverage/20221019/RV32F-20221019/RV32F_SystemVerilog-coverage.log` are machine generated and of very little practical use.  These should not be check into the repo.

**Review by MikeOpenHWGroup — COMMENTED** [review]
Why are the coverage models in files with SVH extensions?   They are not header files, there are primarily covergroups, so I would expect them to be in files with an SV extension.

File: ARVM-FunctionalCoverage/20221102/sv_coverage_CV32E40Pv2/coverage/RISCV_coverage_pkg.svh
Comment (MikeOpenHWGroup) [inline ARVM-FunctionalCoverage/20221102/sv_coverage_CV32E40Pv2/coverage/RISCV_coverage_pkg.svh]: Package files should not have an SVH extension.

**Review by MikeOpenHWGroup — CHANGES_REQUESTED** [review]
Recommend renaming the "coverage" spreadsheets (e.g. `ARVM-FunctionalCoverage/20221019/RV32F-20221019/RV32F_coverage.csv`) to something that indicates they are coverage _specifications_, not coverage implementations.

The core-v-verif nominature for these is "DVplan", so I would suggest `RV32F_DVplan.csv`.

**Comment by simon5656** [comment]
the coverage log files are example of what the coverage actually looks like when run through a simulator -  as the code in SystemVerilog does not show the bins etc and while developing functional coverage it is essential to see things like this

this file is temporary - as was felt useful to be shown and walked through - and as all these files are - and if you like I will just put things into other public repos to save you the review issues... - where is the repo for work in progress... it is this one.
________________________________

**Comment by simon5656** [comment]
they are files that are included... and normally if a file is included it is a termed a header and thus .svh

this repo directory is temporary - as was felt useful to be shown and walked through - and as all these files are - and if you like I will just put things into other public repos to save you the review issues... - where is the repo for work in progress... it is this one.
________________________________

**Comment by simon5656** [comment]
this repo directory is temporary - as was felt useful to be shown and walked through - and as all these files are - and if you like I will just put things into other public repos to save you the review issues... - where is the repo for work in progress... it is this one.

________________________________

**Comment by simon5656** [comment]
From Dave Rich...
The typical convention is that .sv files are placed on the compilers command line, whereas .svh files are read by `include directives from other files

The only files that should be read by `include are macro `defines and design units that are split into multiple files for collaborative source control. For example, each class in a package is typically put into a separate file. You may want to read my post about `include versus import: http://go.mentor.com/package-import-versus-include"
________________________________

**Comment by simon5656** [comment]
"Once the code is in multiple files, we use different file extensions to help us remember if the file needs to be put on the compiler's command line or will be `included from the source of another file. We use the 'h' suffix to indicate a file to be `included mainly because that the convention in many other programming languages. 'h' stands for header, or the prototypes of routines without their implementation. SystemVerilog does not really have the same need for header files, but people still the 'header' terminology to indicate a file that needs to be `included."

With this rationale, (which is good), all the files included in a package should have a .svh extension. In practice, I have seen many cases where people don't do this.
________________________________

**Comment by MikeOpenHWGroup** [comment]
> the coverage log files are example of what the coverage actually looks like when run through a simulator -  as the code in SystemVerilog does not show the bins etc and while developing functional coverage it is essential to see things like this.

Understood and I agree that it is important to be able to generate and review these files.  However, as these files are machine-generated there is no need to have then in a version control repository.  When a developer needs to see the results of a sim they will review the coverage logs they generate, not the coverage logs in a repo generated by someone else.

**Comment by MikeOpenHWGroup** [comment]
> From Dave Rich...
The typical convention is that .sv files are placed on the compilers command line, whereas .svh files are read by `include directives from other files

OK, I'll play ball with that.

**Comment by MikeOpenHWGroup** [comment]
AFAICT, then are only two remaining unresolved points in this review:
1. Recommend renaming the "coverage" spreadsheets (e.g. ARVM-FunctionalCoverage/20221019/RV32F-20221019/RV32F_coverage.csv) to something that indicates they are coverage specifications, not coverage implementations. The core-v-verif nominature for these is "DVplan", so I would suggest RV32F_DVplan.csv.
2. Inclusion of the coverage log files (e.g. `RV32F_SystemVerilog-coverage.log`).

We can deal with the first item in a separate pull-request.   However, the coverage logfile should not be merged in.  Please delete and we are good-to-merge.

**Comment by simon5656** [comment]
yes but until they have some code that works they need to know what is going to be there so these are temporary and for information like lots of other documentation.

________________________________

**Comment by simon5656** [comment]
these .csv files are temporary - to document the development - as we develop the spec of what should be documented we can work on this - for example I think we will not use .csv but write out .md for github for easy of browsing etc and maybe one .xslx for all the extensions in a different tab etc - so again the name is irrelevant - the content is the thing to focus on.

I would far prefer all the comments to be related the work like for example missing coverpoints...


________________________________

**Comment by MikeOpenHWGroup** [comment]
> these .csv files are temporary - to document the development

Ah, I did not understand this.  We should make that clear in a README in the same directory.

> yes but until they have some code that works they need to know what is going to be there so these are temporary and for information like lots of other documentation.

I assume this comment relates to the coverage log file.  There is simply no need for these and dumping large, machine generatable files into GitHub is no useful.

**Comment by silabs-robin** [comment]
> the coverage log files are example of what the coverage actually looks like when run through a simulator

Fair point, that is useful info when browsing a repo.

Could it be reduced to shorter code snippets in a readme?

**Comment by silabs-robin** [comment]
> Recommend renaming the "coverage" spreadsheets (e.g. ARVM-FunctionalCoverage/20221019/RV32F-20221019/RV32F_coverage.csv) to something that indicates they are coverage specifications

> these .csv files are temporary - to document the development

> Ah, I did not understand this. We should make that clear in a README in the same directory.

Is this solved?

_(My 2 cents: if a repo is not structured properly it will just become unnavigable and people abstain from using it. This includes intelligible naming.)_

File: ARVM-FunctionalCoverage/20221201/CV32E40Pv2_riscvISACOV/ChangeLog.md
Comment (MikeOpenHWGroup) [inline ARVM-FunctionalCoverage/20221201/CV32E40Pv2_riscvISACOV/ChangeLog.md]: We cannot accept a copyright file into an OpenHW repo unless it is permissively licensed, ideally with Solderpad-2.0 or Apache-2.- (or newer).

File: ARVM-FunctionalCoverage/20221201/CV32E40Pv2_riscvISACOV/source/RISCV_coverage_config.svh
Comment (MikeOpenHWGroup) [inline ARVM-FunctionalCoverage/20221201/CV32E40Pv2_riscvISACOV/source/RISCV_coverage_config.svh]: Please also add an SPDX header:
```
// SPDX-License-Identifier: Apache-2.0 WITH SHL-2.0
```

For future reference, all that is needed are the Copyright and SPDX statements:
```
// Copyright (c) 2022 Imperas Software Ltd., www.imperas.com
// SPDX-License-Identifier: Apache-2.0 WITH SHL-2.0
```

File: ARVM-FunctionalCoverage/20221201/CV32E40Pv2_riscvISACOV/source/coverage/RISCV_coverage_pkg.svh
Comment (MikeOpenHWGroup) [inline ARVM-FunctionalCoverage/20221201/CV32E40Pv2_riscvISACOV/source/coverage/RISCV_coverage_pkg.svh]: SPDX

File: ARVM-FunctionalCoverage/20221201/CV32E40Pv2_riscvISACOV/source/coverage/RV32F_illegal_coverage.svh
Comment (MikeOpenHWGroup) [inline ARVM-FunctionalCoverage/20221201/CV32E40Pv2_riscvISACOV/source/coverage/RV32F_illegal_coverage.svh]: SPDX

File: ARVM-FunctionalCoverage/20221201/CV32E40Pv2_riscvISACOV/source/coverage/RV32ZCFZFINX_illegal_coverage.svh
Comment (MikeOpenHWGroup) [inline ARVM-FunctionalCoverage/20221201/CV32E40Pv2_riscvISACOV/source/coverage/RV32ZCFZFINX_illegal_coverage.svh]: SPDX

File: ARVM-FunctionalCoverage/20221201/CV32E40S_riscvISACOV/ChangeLog.md
Comment (MikeOpenHWGroup) [inline ARVM-FunctionalCoverage/20221201/CV32E40S_riscvISACOV/ChangeLog.md]: Please add permissive license header.

File: ARVM-FunctionalCoverage/20221201/CV32E40S_riscvISACOV/source/RISCV_coverage_config.svh
Comment (MikeOpenHWGroup) [inline ARVM-FunctionalCoverage/20221201/CV32E40S_riscvISACOV/source/RISCV_coverage_config.svh]: SPDX

File: ARVM-FunctionalCoverage/20221201/CV32E40S_riscvISACOV/source/coverage/RISCV_coverage_common.svh
Comment (MikeOpenHWGroup) [inline ARVM-FunctionalCoverage/20221201/CV32E40S_riscvISACOV/source/coverage/RISCV_coverage_common.svh]: Can we make these `$fatal()`?.

File: ARVM-FunctionalCoverage/20221201/CV32E40S_riscvISACOV/source/coverage/RISCV_coverage_pkg.svh
Comment (MikeOpenHWGroup) [inline ARVM-FunctionalCoverage/20221201/CV32E40S_riscvISACOV/source/coverage/RISCV_coverage_pkg.svh]: SPDX

**Review by MikeOpenHWGroup — CHANGES_REQUESTED** [review]
Looks good, thanks for this!  One serious comment and several minor ones.  The serious comment is that there are at least two files which declare that Imperas holds the copyright, but do not indicate a permissive license.  We cannot accept these into an OpenHW repo.  (I'm sure it was just a simple oversight.)

The other comments are asking for the addition of an SPDX header.

**Comment by silabs-robin** [comment]
@simon5656, @MikeOpenHWGroup, it seems the copyright headers are the only remaining issue blocking this PR?

**Comment by MikeOpenHWGroup** [comment]
> @simon5656, @MikeOpenHWGroup, it seems the copyright headers are the only remaining issue blocking this PR?

That is the big one.  I'd also like to see the $error() calls replaced with $fatal().  If the user misconfigs the compiler macros the whole thing should terminate at t=0.

How quickly do you need this?   Both @simon5656 and I are at the RISC-V Summit this week and otherwise occupid.

**Comment by silabs-robin** [comment]
> > @simon5656, @MikeOpenHWGroup, it seems the copyright headers are the only remaining issue blocking this PR?
> 
> That is the big one. I'd also like to see the $error() calls replaced with $fatal().
> 
> How quickly do you need this?

We would kinda need it now. @MikeOpenHWGroup 

Because afterwards, there will be a job to connect this Zc coverage up in core-v-verif to get some actual metrics out of it.
And after that, there will be a job to process the metrics, potentially tweaking constrained random or waiving.
So any delay before that will just push us too far into the future.

I know it is christmas times, but those changes above are fairly small. @simon5656, could you fix it this week?

**Comment by simon5656** [comment]
fixed error->fatal and SPDX header. in pull request

**Review by MikeOpenHWGroup — APPROVED** [review]

**Comment by MikeOpenHWGroup** [comment]
Thanks for the updates @simon5656.   Merging...

**Comment by silabs-robin** [comment]
Excellent! Thank you both. 🙇‍♂️

**Pull request merged**
